src/lisflood/hydrological_modules/waterabstraction.py

```diff
@@ -268,29 +263,17 @@ def dynamic(self):
             # ***** READ WATER DEMAND DATA *****************************
             # ************************************************************
 
+            # if transient, get step from water demand forcings, else use value in init
             if option['TransientWaterDemandChange']:
                 if option['readNetcdfStack']:
                     # Read using xarray reader
-                    
                     step = self.var.currentTimeStep() - self.var.firstTimeStep()
-
                     self.var.DomesticDemandMM = self.forcings['DomesticDemandMaps'][step] * self.var.DtDay
                     self.var.IndustrialDemandMM = self.forcings['IndustrialDemandMaps'][step] * self.var.DtDay
                     self.var.LivestockDemandMM = self.forcings['LivestockDemandMaps'][step] * self.var.DtDay
                     self.var.EnergyDemandMM = self.forcings['EnergyDemandMaps'][step] * self.var.DtDay
-
                 else:
-                    # Read from stack of maps in Pcraster format
-                    self.var.DomesticDemandMM = readmapsparse(binding['DomesticDemandMaps'], self.var.currentTimeStep(),
-                                                              self.var.DomesticDemandMM) * self.var.DtDay
-                    self.var.IndustrialDemandMM = readmapsparse(binding['IndustrialDemandMaps'],
-                                                                self.var.currentTimeStep(),
-                                                                self.var.IndustrialDemandMM) * self.var.DtDay
-                    self.var.LivestockDemandMM = readmapsparse(binding['LivestockDemandMaps'],
-                                                               self.var.currentTimeStep(),
-                                                               self.var.LivestockDemandMM) * self.var.DtDay
-                    self.var.EnergyDemandMM = readmapsparse(binding['EnergyDemandMaps'], self.var.currentTimeStep(),
-                                                            self.var.EnergyDemandMM) * self.var.DtDay
+                    raise LisfloodError("PCRaster format not supported for the TransientWaterDemandChange option")
```

Comment:
Sorry @doc78, I forgot to answer and now I just got the notification that you merged the branch!

We removed the PCRaster reader because it doesn't really make sense for the transient water demand as PCRaster doesn't have any temporal information. So the only way to use it would be to have all the steps of the run ready in the right order... while we usually have climatology files for the water demand, with monthly values. But maybe you see a use case we missed? We could revise that part if needed, no issues.

Reply:
No worries, we still have this code in the dev and master branch, so we can evaluate it later